awx/main/middleware.py

```diff
@@ -76,63 +72,6 @@ def save_profile_file(self, request):
         return filepath
 
 
-class ActivityStreamMiddleware(threading.local, MiddlewareMixin):
```

Comment:
I'm pretty certain this `threading.local` is responsible for this comment:

https://github.com/ansible/tower/issues/4037#issuecomment-607429832

This middleware is leaking state between requests.